### ISSUE TITLE
Pages.Platform.withFileGenerator makes build fail if I actually perform a StaticHttp.Request

### ISSUE THREAD

**Issue by tgelu** [dillonkearns/elm-pages#107]
Steps to reproduce:

1. clone https://github.com/dillonkearns/elm-pages-starter
2. in `Main.elm` change

```elm
generateFiles siteMetadata =
    StaticHttp.succeed
        [ Feed.fileToGenerate { siteTagline = siteTagline, siteUrl = canonicalSiteUrl } siteMetadata |> Ok
        , MySitemap.build { siteUrl = canonicalSiteUrl } siteMetadata |> Ok
        ]
```

to anything but `StaticHttp.succeed`, like the following

```elm
getRequest : StaticHttp.Request String
getRequest =
    StaticHttp.request
        (Secrets.succeed
            (\token ->
                { url = "http://someapi.com/endpoint"
                , method = "GET"
                , headers =
                    [ ( "Authorization", "Bearer " ++ token ) ]
                , body = StaticHttp.emptyBody
                }
            )
            |> Secrets.with "TOKEN"
        )
        (OptimizedDecoder.field "domain" OptimizedDecoder.string)

generateFiles siteMetadata =
    getRequest
        |> StaticHttp.map
            (\str ->
                [ Feed.fileToGenerate { siteTagline = siteTagline, siteUrl = canonicalSiteUrl } siteMetadata |> Ok
                , MySitemap.build { siteUrl = canonicalSiteUrl } siteMetadata |> Ok
                ]
            )
```

3. Notice that both `elm-pages develop` and `elm-pages build` just hang, no matter if the request fails or succeeds.

PS: I think elm-pages is fantastic. Thank you for creating it!

**Comment by dillonkearns** [comment]
Hello @tgelu, thanks so much for reporting the issue! Good news, I've found the root cause. It was a bit tricky, but fortunately with the help of a little unit testing the fix is not too difficult. The problem is that the new caching functionality I added is stopping the build, but it isn't sending out the data because it doesn't think it's done yet (even though it is since the StaticHttp data is cached).

Hopefully I'll have a fix for this released shortly!
